src/shared/sleep-config.c

```diff
@@ -102,6 +102,42 @@ int parse_sleep_config(SleepConfig **ret_sleep_config) {
         return 0;
 }
 
+bool is_battery_low(void) {
+        _cleanup_free_ char *bat_cap_level = NULL;
+        int r;
+
+        r = read_one_line_file("/sys/class/power_supply/BAT0/capacity_level", &bat_cap_level);
+        if (r < 0) {
+                log_debug_errno(r, "Failed to read /sys/class/power_supply/BAT0/capacity_level: %m");
+                return false;
+        }
+
+        log_debug("Current battery level : %s", bat_cap_level);
+
+        if (STR_IN_SET(bat_cap_level, "Critical", "Low", "Normal"))
+                return true;
+                /* Normal will be removed when estimate battery consumption rate can be estimated to sleep instead of fixed sleep timer. */
+        else
+                return false;
+}
+
+int read_battery_capacity(void) {
+        _cleanup_free_ char *bat_cap = NULL;
+        int battery_capacity, r;
+
+        r = read_one_line_file("/sys/class/power_supply/BAT0/capacity", &bat_cap);
+        if (r < 0)
+               return log_debug_errno(r, "Failed to read /sys/class/power_supply/BAT0/capacity: %m");
+
+        r = safe_atoi(bat_cap, &battery_capacity);
+        if (r < 0)
+               return log_debug_errno(r, "Failed to convert battery capacity char value to int: %m");
+
+        log_debug("Current battery percentage available : %d", battery_capacity);
```

Comment:
maybe adda a `%%` after `%s` given it's a percentage, so that it's clear what it references

Reply:
also extra space before `:`